File: xssfinder-java-executor/src/main/java/org/xssfinder/runner/CustomSubmitterInstantiator.java
package org.xssfinder.runner;

import org.xssfinder.CustomSubmitter;
import org.xssfinder.SubmitAction;
import org.xssfinder.reflection.*;
import org.xssfinder.reflection.InstantiationException;

import java.lang.reflect.Method;

public class CustomSubmitterInstantiator {
    private final Instantiator instantiator;

    public CustomSubmitterInstantiator(Instantiator instantiator) {
        this.instantiator = instantiator;
    }

    public CustomSubmitter instantiate(Method method) {
        SubmitAction submitAction = method.getAnnotation(SubmitAction.class);
        if (submitAction == null) {
            return null;
        }
        Class<? extends CustomSubmitter> submitterClass = submitAction.value();
        if (submitterClass == CustomSubmitter.class) {
            // CustomSubmitter is the default value, and indicates that no explicit custom submitter is given
            return null;
        }
        try {
            return instantiator.instantiate(submitterClass);
        } catch (InstantiationException e) {
            throw new CustomSubmitterInstantiationException(e);
        }
    }
}

File: xssfinder-php-executor/src/XssFinder/Runner/CustomSubmitterInstantiator.php
<?php

namespace XssFinder\Runner;

use ReflectionMethod;
use SubmitActionAnnotation;
use XssFinder\Annotations\Annotations;
use XssFinder\Annotations\CustomSubmitter;
use XssFinder\Annotations\ExactlyOneAnnotationRequiredException;

class CustomSubmitterInstantiator
{
    /**
     * @param ReflectionMethod $method
     * @return CustomSubmitter
     * @throws InvalidCustomSubmitterException
     * @throws NonExistentCustomSubmitterException
     */
    public function instantiate(ReflectionMethod $method)
    {
        $annotation = SubmitActionAnnotation::getSubmitAction($method);
        $submitterClassName = $annotation->submitterClass;

        if (!class_exists($submitterClassName)) {
            throw new NonExistentCustomSubmitterException("Custom submitter class $submitterClassName cannot be found");
        }

        $submitter = new $submitterClassName();

        $customSubmitterInterface = 'XssFinder\Annotations\CustomSubmitter';
        if (!($submitter instanceof $customSubmitterInterface)) {
            throw new InvalidCustomSubmitterException("$submitterClassName is not an instance of $customSubmitterInterface");
        }

        return $submitter;
    }

    public function hasCustomSubmitter(ReflectionMethod $method)
    {
        try {
            $annotation = SubmitActionAnnotation::getSubmitAction($method);
            return $annotation->submitterClass !== null;
        } catch (ExactlyOneAnnotationRequiredException $e) {
            return false;
        }
    }
}

class InvalidCustomSubmitterException extends \Exception {
    function __construct($message) { parent::__construct($message); }
}

class NonExistentCustomSubmitterException extends \Exception {
    function __construct($message) { parent::__construct($message); }
}